ui/views.py

```diff
@@ -292,3 +297,25 @@ def export(request, repo_slug, resource_id):
         )
     except NotFound:
         raise Http404()
+
+
+@login_required
+def serve_media(request, media_path):
+    """
+    View to serve media files in case settings.DEFAULT_FILE_STORAGE
+    is django.core.files.storage.FileSystemStorage
+    """
+    # first check if the user has access to the file
+    file_path = os.path.join(settings.MEDIA_ROOT, media_path)
+    static_asset = get_object_or_404(StaticAsset, asset=media_path)
+    if (RepoPermission.view_repo[0] not in
+            get_perms(request.user, static_asset.course.repository)):
+        raise PermissionDenied()
+    filename = os.path.basename(file_path)
+    response = StreamingHttpResponse(
+        FileWrapper(open(file_path), 1024),
```

Comment:
You can leave out the 1024, it looks like there is a default blocksize of 8192 and we shouldn't override that unless there's a need to